extensions/spark/kyuubi-extension-spark-3-1/src/main/scala/org/apache/kyuubi/sql/KyuubiSparkSQLParser.scala

```diff
@@ -21,19 +21,21 @@ import org.antlr.v4.runtime._
 import org.antlr.v4.runtime.atn.PredictionMode
 import org.antlr.v4.runtime.misc.{Interval, ParseCancellationException}
 import org.apache.spark.sql.AnalysisException
-import org.apache.spark.sql.catalyst.{FunctionIdentifier, TableIdentifier}
+import org.apache.spark.sql.catalyst.{FunctionIdentifier, SQLConfHelper, TableIdentifier}
 import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.catalyst.parser.{ParseErrorListener, ParseException, ParserInterface, PostProcessor}
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.catalyst.trees.Origin
 import org.apache.spark.sql.types.{DataType, StructType}
 
-abstract class KyuubiSparkSQLParserBase extends ParserInterface {
+abstract class KyuubiSparkSQLParserBase extends ParserInterface with SQLConfHelper {
   def delegate: ParserInterface
-  def astBuilder: KyuubiSparkSQLAstBuilderBase
+  def astBuilder: KyuubiSparkSQLAstBuilder
 
   override def parsePlan(sqlText: String): LogicalPlan = parse(sqlText) { parser =>
     astBuilder.visit(parser.singleStatement()) match {
+      case optimize: UnparsedPredicateOptimize =>
+        astBuilder.buildOptimizeStatement(optimize, delegate.parseExpression)
```

Comment:
it seems a little dangerous to use `delegate.parseExpression` since it is named. What's the behavior of such query ? `OPTIMIZE p where event_date = current_date() as c ZORDER BY c1, c2;`

We should ensure the returned expression is not a named expression.

Reply:
Good catch, we need to add some validation to make sure the expression is what we expect.

Reply:
weird.. We should forbid this syntax since Spark also does not support such query.

Reply:
let me add constraints to disallow it.